eox_core/edxapp_wrapper/backends/users_j_v1.py

```diff
@@ -180,6 +190,74 @@ def get_edxapp_user(**kwargs):
     return user
 
 
+def delete_edxapp_user(*args, **kwargs):
+    """
+    Deletes a user from the platform.
+    """
+    msg = None
+    status_code = None
+
+    user = kwargs.get("user")
+    case_id = kwargs.get("case_id")
+    site = kwargs.get("site")
+    is_support_user = kwargs.get("is_support_user")
+
+    signup_sources = user.usersignupsource_set.all()
+    sources = [x.site for x in signup_sources]
+
+    user_response = "The user {username} <{email}> ".format(username=user.username, email=user.email)
+
+    if site and site.upper() in (source.upper() for source in sources):
+        if len(sources) == 1:
+            with transaction.atomic():
+                support_label = '_support' if is_support_user else ''
+                user.email = "{email}{case}.ednx{support}_retired".format(
+                    email=user.email,
+                    case=str(case_id),
+                    support=support_label,
+                )
+                user.save()
+
+                # Add user to retirement queue.
+                UserRetirementStatus.create_retirement(user)
+
+                # Unlink LMS social auth accounts
+                UserSocialAuth.objects.filter(user_id=user.id).delete()
+                # Change LMS password & email
+                user.email = get_retired_email_by_email(user.email)
+                user.save()
+                _set_unusable_password(user)
+
+                # Remove the activation keys sent by email to the user for account activation.
+                Registration.objects.filter(user=user).delete()
+                # Delete OAuth tokens associated with the user.
+                retire_dot_oauth2_models(user)
+
+                # Delete user signup source object
+                signup_sources[0].delete()
+
+                msg = "{user} has been removed".format(user=user_response)
+                status_code = status.HTTP_204_NO_CONTENT
+        else:
+            for signup_source in signup_sources:
+                if signup_source.site.upper() == site.upper():
+                    signup_source.delete()
+
+                    msg = "{user} has more than one signup source. The signup source on the site {site} has been deleted".format(
+                        user=user_response,
+                        site=site,
+                    )
+                    status_code = status.HTTP_200_OK
+
+    msg = "{user} does not have a signup source on the site {site}".format(user=user_response, site=site)
+    status_code = status.HTTP_400_BAD_REQUEST
+    response_dict = {
+        "status": status_code,
+        "message": msg,
+    }
+    return json.dumps(response_dict)
```

Comment:
Now that I see this, is there any need for the json dump? wouldn't the `Response` object take care of that?

Reply:
you're right, this was showing the response in a weird format.